File: SecondYear/any-mesh_AnimeshKumarSingh_2226CSE(AI&ML)1119_2/Week_2/reverseANumber.cpp
long reverseBits(long n) {
    // Write your code here.
    vector<int> bits(32, 0);
    
    for(int i = 0 ; i < 32 ; i++){
        if((n & (long)1 << i) > 0){
            bits[i] = 1;
        }
    }
    for(int i = 0 ; i < 16 ; i++){
        swap(bits[i], bits[31 - i]);
    }

    long ans = 0;

    for(int i = 0 ; i < 32 ; i++){
        if(bits[i] == 1){
            ans += (long)1 << i;
        }
    }
    return ans;
}

File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/week_2/reverseanumber.java

public class reverseanumber {
	public static long reverseBits(long n) {
		// Write your code here
		long reversed = 0;
        int numberOfBits = 32;  // Assuming 32-bit integers

        for (int i = 0; i < numberOfBits; i++) {
    reversed = (reversed << 1) | (n & 1);
            n = n >> 1;
        }

        return reversed;

	}
}
